### PR TITLE
Workload: Better Freeze view mode in order to debug the Space and View behavior

### PR DESCRIPTION
**Pull request by samcake**
- Improve the freeze view mode of the Workload to be able to look at the state of the space without changing the Workload::View. The view is not updated by the Main frustum in that case.

- Better representation of the View circle

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7980/app_12624_PR.apk

File: interface/src/workload/GameWorkloadRenderer.cpp [hunk 1]
```diff
@@ -201,6 +199,32 @@ const gpu::PipelinePointer GameWorkloadRenderItem::getViewsPipeline() {
     }
     return _drawAllViewsPipeline;
 }
+
+const gpu::BufferPointer GameWorkloadRenderItem::getDrawViewBuffer() {
+    if (!_drawViewBuffer) {
+        int numSegments = 64;
+        float angleStep = M_PI * 2.0 / (float)numSegments;
```
Comment (AndrewMeadows) [inline interface/src/workload/GameWorkloadRenderer.cpp]: `2.0` --> `2.0f`

Is M_PI a double?  If so, maybe you should cast it directly to `float`.

File: interface/src/workload/GameWorkloadRenderer.cpp [hunk 2]
```diff
@@ -201,6 +199,32 @@ const gpu::PipelinePointer GameWorkloadRenderItem::getViewsPipeline() {
...
+        std::vector<Vert> verts(numSegments + 1);
+        for (int i = 0; i < numSegments; i++) {
+            float angle = (float)i * angleStep;
+            verts[i].p.x = cos(angle);
+            verts[i].p.y = sin(angle);
+            verts[i].p.z = angle;
+            verts[i].p.w = 1.0;
+        }
+        verts[numSegments] = verts[0];
+        verts[numSegments].p.w = 0.0;
```
Comment (AndrewMeadows) [inline interface/src/workload/GameWorkloadRenderer.cpp]: `0.0` --> `0.0f`

File: interface/src/workload/GameWorkloadRenderer.cpp [hunk 3]
```diff
@@ -221,8 +245,9 @@ void GameWorkloadRenderItem::render(RenderArgs* args) {
     if (_showViews) {
         batch.setPipeline(getViewsPipeline());
 
-        static const int NUM_VERTICES_PER_VIEW = 27;
-        batch.draw(gpu::TRIANGLES, NUM_VERTICES_PER_VIEW * _numAllViews, 0);
+        batch.setUniformBuffer(0, getDrawViewBuffer());
+        batch.draw(gpu::TRIANGLE_STRIP, _numDrawViewVerts * 3 * 2, 0);
```
Comment (AndrewMeadows) [inline interface/src/workload/GameWorkloadRenderer.cpp]: The `3 * 2` value looks magic to me.  I would guess the `3` comes from the fact that we're talking about triangles, but dunno about the `2`.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7980/HighFidelity-Beta-PR12624-3efa92b48eeb5dfb634ea4213085717a134d5420.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7980/HighFidelity-Beta-PR12624-3efa92b48eeb5dfb634ea4213085717a134d5420.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7983/app_12624_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7983/HighFidelity-Beta-PR12624-b922331b0515db5260f94e2d5cb2ba6a144de417.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7983/HighFidelity-Beta-PR12624-b922331b0515db5260f94e2d5cb2ba6a144de417.exe

**Pull request merged**